packages/@o3r/rules-engine/src/engine/operator/operator.interface.ts

```diff
@@ -10,8 +12,12 @@ export interface Operator<LeftExposed = unknown, RightExposed = unknown, LeftSup
   validateLhs?: unknown extends LeftSupported ? (operand: unknown) => boolean : (operand: unknown) => operand is LeftSupported;
   /** Right Hand Value validator function */
   validateRhs?: unknown extends RightSupported ? (operand: unknown) => boolean : (operand: unknown) => operand is RightSupported;
+  /** Operator Fact Values validator function */
+  validateOperatorFactValues?: unknown extends Record<string, Facts> ? (operators: unknown) => boolean : (operators: unknown) => operators is Record<string, Facts>;
   /** Evaluate the values */
-  evaluator: (lhs: LeftSupported, rhs: RightSupported) => boolean;
+  evaluator: (lhs: LeftSupported, rhs: RightSupported, operatorFactValues?: Record<string, Facts>) => boolean;
+  /** List of facts names used inside the operator */
+  dependencies?: string[];
```

Comment:
Should be part of the extracted metadata

Reply:
Please explain why the `factImplicitDependencies` should be part of the extracted metadata. This field is not intended to be used in the CMS.

Reply:
add `factImplicitDependencies` in the metadata so that applications can perform checks when using custom operators with dependencies